tests/system/knowledge_base/qdrant/test_qdrant_knowledge_base.py

```diff
@@ -0,0 +1,323 @@
+import random
+from copy import copy
+from pathlib import Path
+
+import pytest
+from dotenv import load_dotenv
+from qdrant_client import QdrantClient
```

Comment:
Since `qdrant_client` would become an extra dependency, this import will need to be changed to `pytest.importorskip` (or any alternative mechanism that will simply skip testing `QdrantKnowledgeBase` for users who haven't installed the `qdrant` extra).